factories/httpProxyTransportFactory.js

```diff
@@ -0,0 +1,37 @@
+'use strict';
+
+/*jslint latedef:false*/
+
+(function() {
+  angular.module('o19s.splainer-search')
+    .factory('HttpProxyTransportFactory', [
+      'TransportFactory',
+      '$http',
+      'HttpJsonpTransportFactory',
+      HttpProxyTransportFactory
+    ]);
+
+  function HttpProxyTransportFactory(TransportFactory, $http, HttpJsonpTransportFactory) {
+    var Transport = function(options) {
+      TransportFactory.call(this, options);
+    };
+
+    Transport.prototype = Object.create(TransportFactory.prototype);
+    Transport.prototype.constructor = Transport;
+
+    Transport.prototype.query = query;
+
+    function query(url, payload, headers) {
+      var transport = this.options().transport;
+      
+      // It doesn't make sense to use JSONP instead of GET with a proxy
+      if (transport instanceof HttpJsonpTransportFactory) {
+        throw new Error('It does not make sense to proxy a JSONP connection, use GET instead.');
+      }
+      url = this.options().proxyUrl + url;
```

Comment:
Is the "contract" here that a proxy URL has a pattern like `https://host/proxy?url=`? That would be fine but we should always URL-encode the URL if we pass it as a parameter (i.e. using `encodeURIComponent`).